components/storage/blockdevice/COMPONENT_SD/SDBlockDevice.cpp

```diff
@@ -753,7 +753,7 @@ int SDBlockDevice::_cmd(SDBlockDevice::cmdSupported cmd, uint32_t arg, bool isAc
 
     // Select card and wait for card to be ready before sending next command
     // Note: next command will fail if card is not ready
-    _select();
+    _spi.select();
```

Comment:
I'm not convinced you don't need the "fill character" before asserting SSEL for the command.  I don't have the specs, but a bit of Googling suggests that doing something to ensure some clock transitions before asserting SSEL is standard for SDCards in SPI mode. I did find this from a SanDisk document on the command completion:

> After the last SPI bus transaction, the host is required to provide 8 (eight) clock cycles for the card to complete the operation before shutting down the clock. Throughout this 8-clock period, the state of the CS signal is irrelevant. It can be asserted or de-asserted

So I think you will need to retain `_select` and `_deselect` in some form unless you or Seppo know something I don't.

It would be neatest if both can actually be done with SSEL asserted, otherwise you'll need manual lock calls:

It would maybe be most flexible to have helpers like

    void preclock_then_select()
    {
         _spi.lock();
         _spi.write(FILL_CHAR);
         _spi.select();
         _spi.unlock();
    }

    void postclock_then_deselect();
    {
        _spi.write(FILL_CHAR);
        _spi.deselect();
     }

(extend/adapt for select_then_preclock or deselect_then_postclock).

If you write them like that, then you're free to have mismatched pairs like `preclock_then_select` followed by just `deselect` on error - any manual lock is only temporary, not in addition to the selection.

Reply:
Done